src/main/java/org/humancellatlas/ingest/project/Project.java

```diff
@@ -36,29 +38,24 @@ public Project(Object content) {
         super(EntityType.PROJECT, content);
     }
 
-    public MetadataDocument addToSubmissionEnvelopes(SubmissionEnvelope submissionEnvelope) {
+    public void addToSubmissionEnvelopes(@NotNull SubmissionEnvelope submissionEnvelope) {
         this.submissionEnvelopes.add(submissionEnvelope);
-        return this;
     }
 
     @JsonIgnore
     public List<SubmissionEnvelope> getOpenSubmissionEnvelopes(){
         return this.submissionEnvelopes.stream()
-                .filter(env -> env.isOpen())
+                .filter(Objects::nonNull)
```

Comment:
Are we sometimes adding null to the set of submissionEnvelopes, to necessitate this check?

Reply:
As discussed, this can happen if the delete goes wrong. I believe due to the non-transactional nature of mongo.